Assets/MixedRealityToolkit.Providers/XRSDK/Editor/ConfigurationChecker/XRSDKConfigurationChecker.cs

```diff
@@ -0,0 +1,131 @@
+        /// - Save the Microsoft.MixedReality.Toolkit.Providers.XRSDK.asmdef file
+        /// This will result in Unity reloading the assembly with the appropriate dependencies.
+        /// </remarks>
+        private static void UpdateAsmDef()
+        {
+            FileInfo[] asmDefFiles = FileUtilities.FindFilesInAssets(AsmDefFileName);
+
+            if (asmDefFiles.Length == 0)
+            {
+                Debug.LogWarning($"Unable to locate file: {AsmDefFileName}");
+                return;
+            }
+            if (asmDefFiles.Length > 1)
+            {
+                Debug.LogWarning($"Multiple ({asmDefFiles.Length}) {AsmDefFileName} instances found. Modifying only the first.");
+            }
+
+            AssemblyDefinition asmDef = AssemblyDefinition.Load(asmDefFiles[0].FullName);
+            if (asmDef == null)
+            {
+                Debug.LogWarning($"Unable to load file: {AsmDefFileName}");
+                return;
+            }
+
+            List<string> references = new List<string>();
+            if (asmDef.References != null)
+            {
+                references.AddRange(asmDef.References);
+            }
+
+            bool changed = false;
+
+#if UNITY_2019_3_OR_NEWER
+            List<VersionDefine> versionDefines = new List<VersionDefine>();
+            if (asmDef.VersionDefines != null)
+            {
+                versionDefines.AddRange(asmDef.VersionDefines);
+            }
+
+            if (!references.Contains(XRManagementReference))
+            {
+                // Add a reference to the ARFoundation assembly
+                references.Add(XRManagementReference);
+                changed = true; 
+            }
+            if (!references.Contains(ARSubsystemsReference))
+            {
+                // Add a reference to the spatial tracking assembly
+                references.Add(ARSubsystemsReference);
+                changed = true;
+            }
+
+            if (!versionDefines.Contains(XRManagementDefine))
+            {
+                // Add the XRManagement #define
+                versionDefines.Add(XRManagementDefine);
+                changed = true;
+            }
+            if (!versionDefines.Contains(ARSubsystemsDefine))
+            {
+                // Add the ARSubsystems #define
+                versionDefines.Add(ARSubsystemsDefine);
+                changed = true;
+            }
+#else
+            if (references.Contains(XRManagementReference))
+            {
+                // Remove the reference to the spatial tracking assembly
+                references.Remove(XRManagementReference);
+                changed = true;
+            }
+            if (references.Contains(ARSubsystemsReference))
+            {
+                // Add a reference to the spatial tracking assembly
+                references.Remove(ARSubsystemsReference);
+                changed = true;
+            }
+#endif
```

Comment:
remove the defines also?

Reply:
The version defines? They don't exist in asmdefs in Unity 2018, so they should be removed on reserialization automatically.